test/swivel/Swivel.sol

```diff
@@ -499,17 +524,35 @@ contract Swivel {
     return true;
   }
 
-  /// @notice Allows the admin to set a new fee denominator
-  /// @param i The index of the new fee denominator
-  /// @param d The new fee denominator
-  function setFee(uint16 i, uint16 d) external authorized(admin) returns (bool) {
-    if (d < MIN_FEENOMINATOR) {
-      revert Exception(18, uint256(d), 0, address(0), address(0));
+  /// @notice allows the admin to set new fee denominators
+  /// @param f array of length 4 holding values which will replace any at the same index in the current feenominators
+  /// @dev note that, since 0 values are allowable the way to leave a feenominator value unchanged is to pass the existing value
+  function changeFee(uint16[4] calldata f) external authorized(admin) returns (bool) {
+    if (feeChange == 0) {
+      revert Exception(35, 0, 0, address(0), address(0));
     }
 
-    feenominators[i] = d;
+    if (block.timestamp < feeChange) {
+      revert Exception(36, block.timestamp, feeChange, address(0), address(0));
+    }
+
+    for (uint256 i; i < 4;) {
+      if (f[i] < MIN_FEENOMINATOR) {
+        revert Exception(18, f[i], MIN_FEENOMINATOR, address(0), address(0));
+      }
```

Comment:
This seems like `admin` input sanitation. That said, I can definitely see the merits of including it to avoid a fat-finger breaking the protocol for a bit.

Reply:
i agree that the MIN can be done away with. Didn't we remove it from Illuminate?

anyway - a product question that we can come back to - it's been there since we introduced the fees